test/ctia/entity/incident_test.clj

```diff
@@ -779,102 +780,108 @@
-                             [new-status next-time] [["Open" (+sec (+ created new_to_opened))]
-                                                     ["Closed" (+sec (+ created new_to_opened opened_to_closed))]]]
-                       (testing (pr-str [new-status incident])
-                         (let [{:keys [parsed-body] :as response} (helpers/fixture-with-fixed-time
-                                                                    (jt/java-date next-time)
-                                                                    #(post-status app (uri/uri-encode incident-id) new-status))]
-                           (assert (= 200 (:status response))
-                                   (pr-str response))))))
-                 avg #(quot (apply + %&) (count %&))]
+                     (doseq [{:keys [created-incident open-at close-at open-status closed-status]}
+                             specs-with-incidents]
+                       (let [open-resp (helpers/fixture-with-fixed-time (jt/java-date (+sec open-at))
+                                                                        #(post-status app (uri/uri-encode (:id created-incident)) open-status))
+                             close-resp (helpers/fixture-with-fixed-time (jt/java-date (+sec close-at))
+                                                                         #(post-status app (uri/uri-encode (:id created-incident)) closed-status))]
+                         (assert (= 200 (:status open-resp) (:status close-resp))
+                                 "expect status updates to succeed"))))]
              (testing "average aggregation"
                (helpers/fixture-with-fixed-time
                  (jt/java-date (jt/plus epoch (jt/days 1))) ;;default `to` query param
                  (fn []
                    ;; the average of intervals.<field> should involve <expected-count> incidents and
                    ;; have value <expected-average> between time window <from> and <to> (latter is optional).
-                   (doseq [[field expected-count expected-average from to :as test-case]
-                           [["new_to_opened" 3 (avg first-new_to_opened second-new_to_opened third-new_to_opened) first-created]
-                            ["new_to_opened" 2 (avg second-new_to_opened third-new_to_opened) second-created]
-                            ["new_to_opened" 2 (avg first-new_to_opened second-new_to_opened) first-created (inc second-created)]
-                            ["new_to_opened" 1 first-new_to_opened first-created (inc first-created)]
-                            ["new_to_opened" 1 second-new_to_opened second-created (inc second-created)]
-                            ["new_to_opened" 1 third-new_to_opened third-created (inc third-created)]
-                            ["new_to_opened" 1 third-new_to_opened third-created]
-                            ["new_to_opened" 0 nil (inc third-created)]
```

Comment:
Are the `(nil? expected-avg)` cases still covered?

Reply:
Good catch. I think I got a little carried away with refactoring that test. I opted to revert and do a more minimal modification to exercise these new cases.